### ISSUE TITLE
Multiple authors see different content for the entrys "current" in Craft

### ISSUE THREAD

**Issue by lenvanessen** [craftcms/cms#9922]
### Description
I'll try to explain this the best I can. We've run into an issue where if you have simultaniously opened an entry with another author, and are both working on it, Craft loads different revisions for both authors, even after refreshing. 

Usually, there would be an indication of "edited" and the option to discard changes, but these are not visible for us. We've made a video of a side-by-side editing with a colluague to demonstrate:
https://backspace.stackstorage.com/s/avtK2983YMAFoAzY

You can see that no matter what we do, my changes will reflect on the revision that my colleague is working on, but not the other way around, her changes never make it to my revision and dissapear into the database.

### Steps to reproduce

1. Create an article and open up the editor screen with a colleague on a different machine
2. Updated the entry, refresh and check the current version together with that colleague. Both screens will show that they are currently showing the lastest revision, when in reality they are showing different things.

### Additional info

- Craft version: 3.7.14
- PHP version: 8.0
- Database driver & version: MySQL 8.

**Comment by brandonkelly** [comment]
Do you have any plugins or modules that could be interfering with the content?

**Comment by lenvanessen** [comment]
@brandonkelly no, happens with plug-ins disabled as well.

**Comment by brandonkelly** [comment]
@lenvanessen Do you have multiple database servers (primary + replica(s))? Perhaps the replica syncing is delayed?

**Comment by lenvanessen** [comment]
@brandonkelly no, single database server and single webserver (seperates vm's though).

**Comment by lenvanessen** [comment]
Hold on, it seems Cloudflare is running in front of the CMS. Perhaps this caches response generated by Craft.

I'll do some further digging to see if this might be the case.

**Comment by brandonkelly** [comment]
Ahh yeah that’s most likely the problem. You should add an exclusion to ensure nothing under your control panel trigger (`admin`) gets cached.
